app.py

```diff
@@ -15,57 +12,40 @@
 
 class Thing(Resource):
     def get(self, name):
-        # TODO: do not interact with the DAO from the APP, the APP will communicate with the MODEL to handle all business logic
-        thing = dao.find(name)
-        
+        thing = Model.get_by_name(name)
         if not thing:
             abort(404, message='Thing {} not found'.format(name))
-            
-        return thing.serialize()
+        return vars(thing)
 
     def delete(self, name):
-        # TODO: do not interact with the DAO from the APP, the APP will communicate with the MODEL to handle all business logic
-        dao.delete(name)
-        
-        return name, 204
+        thing = Model.delete(name)
+        if not thing:
+            abort(404, message='Thing {} not found'.format(name))
+        return vars(thing), 204
 
     def put(self, name):
         args = parser.parse_args()
-        thing = dao.delete(name)
+        thing = Model.update(name, args['function'])
         if not thing:
             abort(404, message='Thing {} not found'.format(name))
-        
-        # TODO: do not interact with the DAO from the APP, the APP will communicate with the MODEL to handle all business logic
-        dao.add(Model.from_dict({'name':name, 'function':args['function']}))
-
-        return name, 201
-
+        return vars(thing), 201
 
 class ThingList(Resource):
     def get(self):
-        thing_list = list(dao.find_all())
+        thing_list = list(Model.find_all())
         thing_dict = {}
-        for i in range(0,len(thing_list)):
-
-            # TODO: debugging, if i is not within thing_list you will get an error below with [i]['name'] ?
-            print(thing_list[i])
-
-            thing_dict[str(i)] = {'name':thing_list[i]['name'],'function':thing_list[i]['function']}
+        for t in thing_list:
+            thing_dict[str(t['_id'])] = {'name':t['name'],'function':t['function']}
         return thing_dict
 
     def post(self):
         args = parser.parse_args()
-        thing = Model.from_dict({'name':args['name'], 'function':args['function']})
+        thing = Model.save(args['name'],args['function'])
```

Comment:
you need to validate the payload passed in here, it will be a json body with parameters to be saved .. that payload after being validated will be passed directly into the model save method